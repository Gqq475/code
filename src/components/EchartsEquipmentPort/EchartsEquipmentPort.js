import React from 'react'
import classes from './EchartsEquipmentPort.scss'
import Chart from 're-echarts'
import EquipmentPortTable from 'components/Table'

type Props = {
  data: Obejct,
  getEquipmentPort: Function,
};

const field = {
  [_.PATH]: 'path',
  [_.PATHTYPE]: 'PathType',
  [_.DEFECTRATE]: 'DefectRate',
  [_.DEFECTNUMBER]: 'DefectNumber',
  [_.GLASSNUMBER]: 'GlassNumber'
}

const option = {
  tooltip: {
    trigger: 'axis'
  },
  legend: {
    data: ['Defect Number', 'Glass Number', 'Defect Rate']
  },
  xAxis: [
    {
      type: 'category'
    }
  ],
  yAxis: [
    {
      type: 'value',
      name: 'Defect Number',
      axisLabel: {
        formatter: '{value}'
      }
    },
    {
      type: 'value',
      name: 'Defect Rate',
      axisLabel: {
        formatter: '{value} %'
      }
    }
  ],
  series: [
    {
      name: 'Defect Number',
      type: 'bar'
    },
    {
      name: 'Glass Number',
      type: 'bar'
    },
    {
      name: 'Defect Rate',
      type: 'line',
      yAxisIndex: 1
    }
  ]
}

export class EchartsEquipmentPort extends React.Component {
  props: Props;

  componentWillMount () {
    this.props.getEquipmentPort()
  }

  render () {
    const defectNumber = []
    const defectRate = []
    const glassNumber = []
    const pathType = []
    if (this.props.data && this.props.data.equipmentPort) {
      this.props.data.equipmentPort.result.forEach((item, i) => {
        pathType.push(item.PathType)
        defectNumber.push(item.DefectNumber)
        defectRate.push(parseFloat(item.DefectRate.substr(0, item.DefectRate.length - 1).replace(',', '')))
        glassNumber.push(item.GlassNumber)
      })
      option.xAxis[0].data = pathType
      option.series[0].data = defectNumber
      option.series[2].data = defectRate
      option.series[1].data = glassNumber
    }

    return (
      <div className={classes['EchartsEquipmentPort-container']}>
        <Chart option={option} style={{height: '500px'}} />
        {
          this.props.data.equipmentPort &&
            <EquipmentPortTable field={field} data={this.props.data.equipmentPort.result} />
        }
      </div>
    )
  }
}

export default EchartsEquipmentPort
