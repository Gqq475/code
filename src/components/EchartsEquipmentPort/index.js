import EchartsEquipmentPort from './EchartsEquipmentPort'
export default EchartsEquipmentPort
