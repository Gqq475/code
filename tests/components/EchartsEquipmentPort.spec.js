import React from 'react'
import { EchartsEquipmentPort } from 'components/EchartsEquipmentPort/EchartsEquipmentPort'

describe('(Component) EchartsEquipmentPort', () => {
  it('render normally', () => {
    const data = {
      equipmentPort: {
        result: [{
          DefectNumber: 62,
          DefectRate: '885.71%',
          GlassNumber: 7,
          PathType: 76,
          path: undefined
        }]
      }
    }
    const getEquipmentPort = jest.fn()
    const wrapper = shallow(
      <EchartsEquipmentPort
        data={data}
        getEquipmentPort={getEquipmentPort} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
