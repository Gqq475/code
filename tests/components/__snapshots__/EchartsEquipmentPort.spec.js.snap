// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) EchartsEquipmentPort render normally 1`] = `
<div
  className="EchartsEquipmentPort-container"
>
  <ECharts
    config={Object {}}
    id="chart"
    lazyUpdate={false}
    notMerge={false}
    option={
      Object {
        "legend": Object {
          "data": Array [
            "Defect Number",
            "Glass Number",
            "Defect Rate",
          ],
        },
        "series": Array [
          Object {
            "data": Array [
              62,
            ],
            "name": "Defect Number",
            "type": "bar",
          },
          Object {
            "data": Array [
              7,
            ],
            "name": "Glass Number",
            "type": "bar",
          },
          Object {
            "data": Array [
              885.71,
            ],
            "name": "Defect Rate",
            "type": "line",
            "yAxisIndex": 1,
          },
        ],
        "tooltip": Object {
          "trigger": "axis",
        },
        "xAxis": Array [
          Object {
            "data": Array [
              76,
            ],
            "type": "category",
          },
        ],
        "yAxis": Array [
          Object {
            "axisLabel": Object {
              "formatter": "{value}",
            },
            "name": "Defect Number",
            "type": "value",
          },
          Object {
            "axisLabel": Object {
              "formatter": "{value} %",
            },
            "name": "Defect Rate",
            "type": "value",
          },
        ],
      }
    }
    style={
      Object {
        "height": "500px",
      }
    }
  />
  <UniverseTable
    data={
      Array [
        Object {
          "DefectNumber": 62,
          "DefectRate": "885.71%",
          "GlassNumber": 7,
          "PathType": 76,
          "path": undefined,
        },
      ]
    }
    field={
      Object {
        "undefined": "GlassNumber",
      }
    }
  />
</div>
`;
